File: solution/kotlin/day13/src/main/kotlin/blog/Blog.kt
package blog

import java.time.LocalDate

class Article(
    private val name: String,
    private val content: String,
    val comments: MutableList<Comment> = mutableListOf()
) {
    private fun addComment(
        text: String,
        author: String,
        creationDate: LocalDate
    ) {
        val comment = Comment(text, author, creationDate)
        if (comments.contains(comment)) {
            throw CommentAlreadyExistException()
        } else comments.add(comment)
    }

    fun addComment(text: String, author: String) = addComment(text, author, LocalDate.now())
}


data class Comment(val text: String, val author: String, val creationDate: LocalDate)
class CommentAlreadyExistException : Exception()

File: exercise/c#/Day04/Day04/Blog.cs
namespace Day04
{
    public class Article(string name, string content)
    {
        public List<Comment> Comments { get; } = [];

        private void AddComment(
            string text,
            string author,
            DateOnly creationDate)
        {
            var comment = new Comment(text, author, creationDate);
            if (Comments.Contains(comment))
            {
                throw new CommentAlreadyExistException();
            }
            else Comments.Add(comment);
        }

        public void AddComment(string text, string author)
            => AddComment(text, author, DateOnly.FromDateTime(DateTime.Now));
    }

    public record Comment(string Text, string Author, DateOnly CreationDate);

    public class CommentAlreadyExistException : ArgumentException;
}